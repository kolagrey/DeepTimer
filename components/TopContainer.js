import React from 'react';
import { SafeAreaView } from 'react-native';
import styles from '../styles';

const TopContainer = (props) => {
    return (
        <SafeAreaView style={styles.topContainer}>
            {props.children}
        </SafeAreaView>
    );
};


export default TopContainer;