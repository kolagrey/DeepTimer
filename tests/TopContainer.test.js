import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import TopContainer from '../components/TopContainer';

describe('<TopContainer/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<TopContainer />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'RCTSafeAreaView'", () => {
        const tree = renderer
            .create(
                <TopContainer />,
            ).toJSON();
        expect(tree.type).toBe('RCTSafeAreaView');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <TopContainer>
                    <Text>
                        Hello Test
                    </Text>
                </TopContainer>,
            ).toJSON();
        expect(tree.children.length).toBe(1)
    });


});