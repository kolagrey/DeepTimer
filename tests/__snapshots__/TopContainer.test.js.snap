// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopContainer/> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "marginTop": 50,
      "padding": 20,
    }
  }
/>
`;
